Examples/LambdaFunctions/Sources/LambdaAPI/ProductLambdaHandler.swift

```diff
@@ -0,0 +1,222 @@
+                    let value = APIGateway.V2.Response(with: result, statusCode: .ok)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .notFound)
+                    return context.eventLoop.makeSucceededFuture(value)
+                }
+            }
+            return read
+        case .update:
+            logger.info("update")
+            let update = UpdateLambdaHandler(service: service).handle(context: context, event: event)
+                .flatMap { response -> EventLoopFuture<APIGateway.V2.Response> in
+                switch response {
+                case .success(let result):
+                    let value = APIGateway.V2.Response(with: result, statusCode: .ok)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .notFound)
+                    return context.eventLoop.makeSucceededFuture(value)
+                }
+            }
+            return update
+        case .delete:
+            logger.info("delete")
+            let delete = DeleteUpdateLambdaHandler(service: service).handle(context: context, event: event)
+                .flatMap { response -> EventLoopFuture<APIGateway.V2.Response> in
+                switch response {
+                case .success(let result):
+                    let value = APIGateway.V2.Response(with: result, statusCode: .ok)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .notFound)
+                    return context.eventLoop.makeSucceededFuture(value)
+                }
+            }
+            return delete
+        case .list:
+            logger.info("list")
+            let list = ListUpdateLambdaHandler(service: service).handle(context: context, event: event)
+                .flatMap { response -> EventLoopFuture<APIGateway.V2.Response> in
+                switch response {
+                case .success(let result):
+                    let value = APIGateway.V2.Response(with: result, statusCode: .ok)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .forbidden)
+                    return context.eventLoop.makeSucceededFuture(value)
+                }
+            }
+            return list
+        case .unknown:
+            logger.info("unknown")
+            let value = APIGateway.V2.Response(with: APIError.handlerNotFound, statusCode: .forbidden)
+            return context.eventLoop.makeSucceededFuture(value)
+        }
+    }
+    
+    struct CreateLambdaHandler {
```

Comment:
consider making these into functions, e.g. (did not try to compile, so maybe syntax errors):

```swift
func createHandler(service: ProductService, context: Lambda.Context, event: APIGateway.V2.Request) -> EventLoopFuture<Product> {
  guard let product: Product = try? event.object() else {
    return context.eventLoop.makeFailedFuture(APIError.invalidRequest)
  }
  return service.createItem(product: product)
}
```

Reply:
Refactored in 8649522